Method not implemented

<!-- Please search existing issues to avoid creating duplicates. -->

<!-- Use Help > Report Issue to prefill some of these. -->

Steps to Reproduce:

1. Attempt to create a PR on github.dev
2. 🐛 See method not implemented

```
notificationsAlerts.ts:40  Error: Method not implemented.
	at Pn.diffBetween (extension.js:123:5276)
	at dy.gitFiles (extension.js:1948:2230)
	at ay.getGitChildren (extension.js:1931:7742)
	at ay.getChildren (extension.js:1931:6659)
	at v.Y (/microsoft/vs/workbench/api/worker/extensionHostWorker.js:80:16344)
	at v.getChildren (/microsoft/vs/workbench/api/worker/extensionHostWorker.js:80:12712)
	at c.$getChildren (/microsoft/vs/workbench/api/worker/extensionHostWorker.js:80:9275)
	at o.S (/microsoft/vs/workbench/api/worker/extensionHostWorker.js:84:48583)
	at o.Q (/microsoft/vs/workbench/api/worker/extensionHostWorker.js:84:48349)
	at o.M (/microsoft/vs/workbench/api/worker/extensionHostWorker.js:84:47426)
	at o.L (/microsoft/vs/workbench/api/worker/extensionHostWorker.js:84:46637)
	at R.eval [as value] (/microsoft/vs/workbench/api/worker/extensionHostWorker.js:84:45294)
	at a.y (workerMain.js:32:1902)
	at a.fire (workerMain.js:32:2119)
	at r.port1.onmessage (/microsoft/vs/workbench/api/worker/extensionHostWorker.js:91:3801)
```

--- a/src/github/createPRViewProviderNew.ts
+++ b/src/github/createPRViewProviderNew.ts
@@ -699,7 +699,7 @@ export class CreatePullRequestViewProviderNew extends WebviewViewBase implements
 			try {
 				let commits: string[];
 				let patches: string[];
-				if (this.model.compareHasUpstream) {
+				if (await this.model.getCompareHasUpstream()) {
 					commits = (await this.model.gitHubCommits()).map(commit => commit.commit.message);
 					patches = (await this.model.gitHubFiles()).map(file => file.patch ?? '');
 				} else {

--- a/src/view/compareChangesTreeDataProvider.ts
+++ b/src/view/compareChangesTreeDataProvider.ts
@@ -178,7 +178,7 @@ abstract class CompareChangesTreeProvider implements vscode.TreeDataProvider<Tre
 
 	async getChildren(element?: TreeNode) {
 		try {
-			if (this.model.compareHasUpstream) {
+			if (await this.model.getCompareHasUpstream()) {
 				return this.getGitHubChildren(element);
 			} else {
 				return this.getGitChildren(element);
@@ -258,7 +258,7 @@ class CompareChangesFilesTreeProvider extends CompareChangesTreeProvider {
 			if (diff.length === 0) {
 				(this.view as vscode.TreeView2<TreeNode>).message = new vscode.MarkdownString(vscode.l10n.t('There are no commits between the base `{0}` branch and the comparing `{1}` branch', this.model.baseBranch, this.model.getCompareBranch()));
 				return [];
-			} else if (!this.model.compareHasUpstream) {
+			} else if (!(await this.model.getCompareHasUpstream())) {
 				const message = new vscode.MarkdownString(vscode.l10n.t({ message: 'Branch `{0}` has not been pushed yet. [Publish branch](command:git.publish) to see all changes.', args: [this.model.getCompareBranch()], comment: "{Locked='](command:git.publish)'}" }));
 				message.isTrusted = { enabledCommands: ['git.publish'] };
 				(this.view as vscode.TreeView2<TreeNode>).message = message;

--- a/src/view/createPullRequestDataModel.ts
+++ b/src/view/createPullRequestDataModel.ts
@@ -105,7 +105,8 @@ export class CreatePullRequestDataModel {
 		return this._compareHasUpstream;
 	}
 
-	public get compareHasUpstream(): boolean {
+	public async getCompareHasUpstream(): Promise<boolean> {
+		await this._constructed;
 		return this._compareHasUpstream;
 	}
 

